### PR TITLE
[ARO-5445] DNS checker logging custom DNS server as an error

### PR DESCRIPTION
**Pull request by rolandmkunkel**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-5445

### What this PR does / why we need it:

Customer reported ARO operator errors in the logs when custom DNS servers are configured in forward plugin:
https://portal.microsofticm.com/imp/v3/incidents/incident/469705806/summary

This is a [supported setup as per our docs](https://learn.microsoft.com/en-us/azure/openshift/dns-forwarding) and should not be logged as an error.

The DNS checks had been setup initially to be included as cluster conditions for SRE awareness (this did not log the status conditions as errors):
https://github.com/Azure/ARO-RP/blob/3e69b7e742af5aeee471a8d781d518c72fd235dd/pkg/operator/controllers/checker/clusterdns.go

A refactoring was done [in this commit](https://github.com/Azure/ARO-RP/commit/340ebb161f590eaf4ee069d519e6cfdd9d5c5f01). Since then, when the status now needs to be updated, the new logic is returning an [Error to the Reconciler](https://github.com/Azure/ARO-RP/blob/340ebb161f590eaf4ee069d519e6cfdd9d5c5f01/pkg/operator/controllers/checkers/clusterdnschecker/checker.go#L50), which in turn always [returns the error](https://github.com/Azure/ARO-RP/blob/340ebb161f590eaf4ee069d519e6cfdd9d5c5f01/pkg/operator/controllers/checkers/clusterdnschecker/controller.go#L82) to K8s controller-runtime, which is where it is being [logged as an error](https://github.com/Azure/ARO-RP/blob/master/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go#L317).

### CODE REVIEW

File: pkg/operator/controllers/checkers/clusterdnschecker/controller_test.go [hunk 1]
```diff
@@ -39,24 +39,43 @@ func TestReconcile(t *testing.T) {
...
 		{
-			name:                 "no errors",
+			name: "no errors",
+			checkerReturnResult: result{
+				pass:    true,
+				message: "no in-cluster upstream DNS servers",
+			},
 			wantConditionStatus:  operatorv1.ConditionTrue,
-			wantConditionMessage: "No in-cluster upstream DNS servers",
+			wantConditionMessage: "no in-cluster upstream DNS servers",
```
Comment (mociarain) [inline pkg/operator/controllers/checkers/clusterdnschecker/controller_test.go:55]: Our log convention is to start with lower case?

File: pkg/operator/controllers/checkers/clusterdnschecker/checker.go [hunk 2]
```diff
@@ -14,7 +14,12 @@ import (
 )
 
 type clusterDNSChecker interface {
-	Check(ctx context.Context) error
+	Check(ctx context.Context) (result, error)
+}
+
+type result struct {
+	pass    bool
```
Comment (mociarain) [inline pkg/operator/controllers/checkers/clusterdnschecker/checker.go:21]: Weak suggestion for `isSuccess` or just `success` instead of `pass`.

**Review by mociarain — APPROVED** [review]
LGTM. Small suggestion and a question but nothing blocking.

File: pkg/operator/controllers/checkers/clusterdnschecker/checker.go (hunk 2 above)
Comment (rolandmkunkel) [reply pkg/operator/controllers/checkers/clusterdnschecker/checker.go]: done

**Review by mociarain — APPROVED** [review]

**Comment by rolandmkunkel** [comment]
superseded by https://github.com/Azure/ARO-RP/pull/3453 
E2E won't run from fork

**Pull request closed without merging**